rpc/src/rpc_impl.rs

```diff
@@ -266,7 +265,8 @@ impl<S: Storage + Send + core::marker::Sync + 'static> RpcFunctions for RpcImpl<
                     transaction,
                 };
 
-                if let Ok(inserted) = self.memory_pool()?.lock().insert(&storage, entry) {
+                // this block_on will halt the tokio worker until insert completion -- can cause problems if not in a multi-threaded environment (tests)
+                if let Ok(inserted) = futures::executor::block_on(self.memory_pool()?.insert(&storage, entry)) {
```

Comment:
do we have a test for this call? I'm worried it might panic with the current RPC server impl - we should test it first (could be with a regular node) to make sure it doesn't happen

Reply:
ok, there is a test, but it uses the internal testing machinery of `jsonrpc`; we should also test it on a regular node

Reply:
I just checked with a different RPC (`disconnect_protected`), and using `block_on` makes it hang, presumably due to the RPC server's `tokio 0.2` executor being single-threaded. We can't reliably use `await` in our RPC (unless via a direct `Handle`) until https://github.com/AleoHQ/snarkOS/pull/759 goes in.

Reply:
yes. the test was what prompted this comment. would be good to test in staging definitely. the right solution is to not use synchronous jsonrpc.